apps/src/gamelab/animationListModule.js

```diff
@@ -687,28 +688,42 @@ function loadPendingFramesFromSource(key, props, callback) {
   };
 }
 
+
 /**
  * Given a key/serialized-props pair for an animation, work out where to get
  * the spritesheet.
  * @param {!AnimationKey} key
- * @param {!SerializedAnimationProps} props
+ * @param {object} animationList - Used to access props and check if src url
+ *        exists in this project and is deleted
+ * @param {string} channelId - Used to differentiate library animations from others
  * @returns {string}
  */
-export function animationSourceUrl(key, props) {
-  // TODO: (Brad) We want to get to where the client doesn't know much about
-  //       animation versions, by switching to Chris' new Files API.
-  //       in the meantime, be able to request versions only when we export
-  //       JSON for levelbuilders to use.
-
-  // 1. If the animation has a sourceUrl it's external (from the library
-  //    or some other outside source, not the animation API) - and we may need
-  //    to run it through the media proxy.
-  if (props.sourceUrl) {
+export function animationSourceUrl(key, animationList, channelId = "") {
+  const props = animationList.propsByKey[key];
+
+  // If the animation has a sourceUrl it's external (from the library,
+  // a levelbuilder, or an uploaded image) - and we may need to run it through
+  // the media proxy.
+  // ChannelId differentiates anims included by levelbuilders from project animations
+  const fromLibraryOrLevelbuilder = props.sourceUrl && !props.sourceUrl.includes('v3/animations/' + channelId);
+  const uploadedAnimation = props.sourceUrl && props.sourceUrl.includes('v3/animations/' + channelId);
+
+  // No need to append a version for animations in the library/from levelbuilders
+  // or if the srcUrl points to an uploaded animation and contains a version
+  const versionNotNeeded = fromLibraryOrLevelbuilder || (uploadedAnimation && props.sourceUrl.includes('?version'));
+
+  if (versionNotNeeded) {
     return assetPrefix.fixPath(props.sourceUrl);
+  } else if (uploadedAnimation) {
+    if (props.version) {
```

Comment:
@islemaster - in previous commits I was looking at whether the animation was deleted or not. But now it seems like whether or not it's deleted, if it has a props.version we want to use that, otherwise, we just get the latest version. Accordingly, I cut out the 'deleted' check. Are there cases that this simplified check will miss?

Reply:
I don't think this will miss any cases.  From the client's point of view, if an animation is deleted then it's not in the manifest and we should never call this method.  On the other hand if the client sees the animation but the server thinks the animation as deleted, then the server needs to handle it (which it does now).